src/main/java/org/opensearch/performanceanalyzer/collectors/telemetry/RTFNodeStatsAllShardsMetricsCollector.java

```diff
@@ -0,0 +1,406 @@
+            currentPerShardStats.clear();
+        }
+        currentShards.clear();
+        currentShards = Utils.getShards();
+    }
+
+    private static final Map<String, ValueCalculator> maps =
+            new HashMap<String, ValueCalculator>() {
+                {
+                    put(
+                            AllMetrics.ShardStatsValue.INDEXING_THROTTLE_TIME.toString(),
+                            (shardStats) ->
+                                    shardStats
+                                            .getStats()
+                                            .getIndexing()
+                                            .getTotal()
+                                            .getThrottleTime()
+                                            .millis());
+
+                    put(
+                            AllMetrics.ShardStatsValue.CACHE_QUERY_HIT.toString(),
+                            (shardStats) -> shardStats.getStats().getQueryCache().getHitCount());
+                    put(
+                            AllMetrics.ShardStatsValue.CACHE_QUERY_MISS.toString(),
+                            (shardStats) -> shardStats.getStats().getQueryCache().getMissCount());
+                    put(
+                            AllMetrics.ShardStatsValue.CACHE_QUERY_SIZE.toString(),
+                            (shardStats) ->
+                                    shardStats.getStats().getQueryCache().getMemorySizeInBytes());
+
+                    put(
+                            AllMetrics.ShardStatsValue.CACHE_FIELDDATA_EVICTION.toString(),
+                            (shardStats) -> shardStats.getStats().getFieldData().getEvictions());
+                    put(
+                            AllMetrics.ShardStatsValue.CACHE_FIELDDATA_SIZE.toString(),
+                            (shardStats) ->
+                                    shardStats.getStats().getFieldData().getMemorySizeInBytes());
+
+                    put(
+                            AllMetrics.ShardStatsValue.CACHE_REQUEST_HIT.toString(),
+                            (shardStats) -> shardStats.getStats().getRequestCache().getHitCount());
+                    put(
+                            AllMetrics.ShardStatsValue.CACHE_REQUEST_MISS.toString(),
+                            (shardStats) -> shardStats.getStats().getRequestCache().getMissCount());
+                    put(
+                            AllMetrics.ShardStatsValue.CACHE_REQUEST_EVICTION.toString(),
+                            (shardStats) -> shardStats.getStats().getRequestCache().getEvictions());
+                    put(
+                            AllMetrics.ShardStatsValue.CACHE_REQUEST_SIZE.toString(),
+                            (shardStats) ->
+                                    shardStats.getStats().getRequestCache().getMemorySizeInBytes());
+                }
+            };
+
+    private static final ImmutableMap<String, ValueCalculator> valueCalculators =
```

Comment:
We can use ImmmutableMap directly instead of first creating the Map and then clone.